### PR TITLE
Feature/eu seller onboarding

### PR DESCRIPTION
**Pull request by sam-johnson-cko**
# Updating Accounts API to support new fields for EU seller onboarding

[//]: # 'Please add your description here'
* Added `place_of_birth` field to `post`/`put` requests and `get` response for company representative and individual.
* Added `roles` field to `post`/`put` requests and `get` response for company representative.
* Added `financial_details` field to `post`/`put` requests and `get` response for company.
* Added `business_type` field to `post`/`put` requests and `get` response for company.

## Checklist

- [X] Added a changelog entry to the `changelog.md` file in either `abc_spec` or `nas_spec`
- [X] Contacted the Tech Docs team about any corresponding guides that need to be updated for www.checkout.com/docs

### CODE REVIEW

**Comment by chrisi-webster-cko** [comment]
Had a creep, and for the front end, we said we weren't going to show the threshold values that meant documents were needed. Just because some people may just provide the lower numbers to get sub-entities onboarded faster. Is there not a way to just make these requirements due? Or provide an error message?

**Comment by sam-johnson-cko** [comment]
> Had a creep, and for the front end, we said we weren't going to show the threshold values that meant documents were needed. Just because some people may just provide the lower numbers to get sub-entities onboarded faster. Is there not a way to just make these requirements due? Or provide an error message?

If you onboard a sub-entity through the API and they provide values for the financial details that exceed the thresholds then the sub-entity will have requirements due for `bank_statement` or `financial_statement`. Shall I remove the threshold values from the documentation?

File: nas_spec/changelog.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 | Date       | Description of change                                                                                                 |
 | ---------- | --------------------------------------------------------------------------------------------------------------------- |
+| 2022/07/12 | Accounts API Eu Seller Onboarding                                                                                     |
```
Comment (chrisi-webster-cko) [inline nas_spec/changelog.md:5]:
```suggestion
| 2022/07/12 | Adds fields required EU sellers using the Accounts API                                                                                 |
```

File: nas_spec/components/schemas/Platforms/EntityFinancialDocuments.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+type: object
+title: FinancialDocuments
+properties:
+  bank_statement:
+    description: 'Bank statement document is mandatory for average_transaction_value greater than 250000 (minor units) or highest_transaction_value greater than 500000 (minor units). Document Type should be: bank_statement.'
```
Comment (chrisi-webster-cko) [inline nas_spec/components/schemas/Platforms/EntityFinancialDocuments.yaml:5]:
```suggestion
    description: 'Bank statement document becomes mandatory depending on the answer provided for `average_transaction_value` and `highest_transaction_value`. The status of your sub-entity will change to `requirements_due` stating when this is necessary.
```

File: nas_spec/components/schemas/Platforms/EntityFinancialDocuments.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    allOf:
+      - $ref: '#/components/schemas/EntityDocument'
+  financial_statement:
+    description: 'Financial statement document is mandatory for annual_processing_volume greater than 1000000000 (minor units). Document Type should be: financial_statement.'
```
Comment (chrisi-webster-cko) [inline nas_spec/components/schemas/Platforms/EntityFinancialDocuments.yaml:9]:
```suggestion
    description: 'Financial statement document becomes mandatory depdning on the answer provided for `annual_processing_volume`. The status of your sub-entity will change to `requirements_due` stating when this is necessary.
```

**Review by chrisi-webster-cko — COMMENTED** [review]
Would confirm with one of the PMs about what I'm changing here, but that's why we've omitted it from the checkout.coms/docs site.

**Review by chrisi-webster-cko — DISMISSED** [review]

**Pull request merged**